WordPressKit/WordPressComOAuthClient.swift

```diff
@@ -444,59 +442,71 @@ public final class WordPressComOAuthClient: NSObject {
         )
     }
 
+    /// Request a security key challenge from WordPress.com to be signed by the client.
+    ///
+    /// - Parameters:
+    ///     - userID: the wpcom userID
+    ///     - twoStepNonce: The nonce returned from a log in attempt.
+    public func requestWebauthnChallenge(
+        userID: Int64,
+        twoStepNonce: String
+    ) async -> WordPressAPIResult<WebauthnChallengeInfo, AuthenticationFailure> {
+        let builder = webAuthnRequestBuilder(action: .requestChallenge)
+            .body(form: [
+                "user_id": "\(userID)",
+                "client_id": clientID,
+                "client_secret": secret,
+                "auth_type": "webauthn",
+                "two_step_nonce": twoStepNonce,
+            ])
+        return await webAuthnSession
+            .perform(request:  builder)
+            .mapUnacceptableStatusCodeError(AuthenticationFailure.init(response:body:))
+            .mapSuccess { response in
+                // Expect the parent data response object
+                guard let responseObject = try? JSONSerialization.jsonObject(with: response.body),
+                      let responseDictionary = responseObject as? [String: Any],
+                      let responseData = responseDictionary["data"] as? [String: Any] else {
+                    return nil
+                }
+
+                // Expect the challenge info.
+                guard
+                    let challenge = responseData["challenge"] as? String,
+                    let nonce = responseData["two_step_nonce"] as? String,
+                    let rpID = responseData["rpId"] as? String,
+                    let allowCredentials = responseData["allowCredentials"] as? [[String: Any]]
+                else {
+                    return nil
+                }
```

Comment:
Did you consider making `Decodable` `struct`s to handle this parsing?

Reply:
I worry "reverse engineering" this parsing code would cause issues...I try to avoid changing too much during this refactor.

Reply:
Fair enough. It's not like we can't come back later and do it 👍